### ISSUE TITLE
Add ability to retain newlines when calling getData()

### ISSUE THREAD

**Issue by mortenson** [ckeditor/ckeditor5#11832]
## 📝 Provide a description of the improvement

In CKEditor 4, newlines (`\n` or `\r\n`) would be retained when getting the HTML representation of CKEditor data.

When calling `getData()` in CKEditor 5, there appear to be no newlines. This seems like a cosmetic detail - but when storing this data in a format that displays newlines, for example in Yaml multiline strings, newlines allow the data to be represented in a nice way and edited directly in the Yaml file.

It would be nice if there was a way, even if it's optional, to get an HTML string out that includes newlines. Thank you!

---

If you'd like to see this improvement implemented, add a 👍 reaction to this post.

**Comment by mortenson** [comment]
Someone in Drupal Slack suggested that pretty printing could be an option and I think that'd be great - pretty printing has the added bonus of making the output HTML more readable for debugging, so it seems useful even outside of my very specific use case. Pretty printing of course could be done by the client, but if CKEditor is willing to put it in I support that!

**Comment by Reinmar** [comment]
I talked about this with @wimleers to understand whether it's more about preserving the original formatting or pretty-printing the output.

The former (preserving insignificant white-spaces) isn't feasible.

The latter is relatively easy. It could be an option in `getData()` and that's what I wanted to propose. But then I realized that it's actually completely doable by an external tool since this can be done without interacting with any editor's internals.

So, I'd recommend going with that for now
